nucleus/payara-modules/notification-core/src/main/java/fish/payara/nucleus/notification/admin/GetNotificationConfiguration.java

```diff
@@ -91,32 +88,46 @@ public void execute(AdminCommandContext context) {
 
         String headers[] = {"Enabled", "Notifier Enabled", "Notifier Name"};
         ColumnFormatter columnFormatter = new ColumnFormatter(headers);
-        Object values[] = new Object[3];
-        values[0] = notificationServiceConfiguration.getEnabled();
 
-        for (ServiceHandle<BaseNotifierService> serviceHandle : allServiceHandles) {
-
-            NotifierConfiguration notifierConfiguration = configuration.getNotifierConfigurationByType(serviceHandle.getService().getNotifierConfigType());
-            LogNotifierConfiguration logNotifierConfiguration = (LogNotifierConfiguration) notifierConfiguration;
-            if (logNotifierConfiguration == null) {
-                values[1] = "true";
-            } else {
-                values[1] = logNotifierConfiguration.getEnabled();
+        if (configuration.getNotifierConfigurationList().isEmpty()) {
+            mainActionReport.setMessage("No notifier defined");
+        }
+        else {
+            List<Class<NotifierConfiguration>> notifierConfigurationClassList = Lists.transform(configuration.getNotifierConfigurationList(), new Function<NotifierConfiguration, Class<NotifierConfiguration>>() {
+                @Override
+                public Class<NotifierConfiguration> apply(NotifierConfiguration input) {
+                    return resolveNotifierConfigurationClass(input);
+                }
+            });
+
+            Properties extraProps = new Properties();
+            for (ServiceHandle<BaseNotifierService> serviceHandle : allServiceHandles) {
+                NotifierConfiguration notifierConfiguration = configuration.getNotifierConfigurationByType(serviceHandle.getService().getNotifierConfigType());
+                if (notifierConfigurationClassList.contains(resolveNotifierConfigurationClass(notifierConfiguration))) {
```

Comment:
This needs to be changed to something like this to prevent you getting a null pointer exception from the `resolveNotifierConfigurationClass(notifierConfiguration)` method:

`if (notifierConfiguration != null && notifierConfigurationClassList.contains(resolveNotifierConfigurationClass(notifierConfiguration))) {`